### PR TITLE
Add JupyterLab Git documentation

### PR DESCRIPTION
**Pull request by kouzant**

### CODE REVIEW

File: docs/user_guide/hopsworks/jupyter.rst [hunk 1]
```diff
@@ -216,6 +216,92 @@ When you download a dataframe from spark to pandas with sparkmagic, it gives you
     Plot a local pandas dataframe using matplotlib and the magic %%local
 
 
+Jupyter notebooks with version control
+--------------------------------------
+
+Jupyter notebooks have become the lingua franca for data scientists. As with ordinary source code files, we should version them
+to be able to keep track of the changes we made or collaborate.
+
+**Hopsworks Enterprise Edition** comes with a feature to allow users version their notebooks with Git and interact with remote repositories such as
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/jupyter.rst]: @kouzant `to allow users to version`

Comment (tkakantousis) [inline docs/user_guide/hopsworks/jupyter.rst]: @kouzant `to allow users to version`

File: docs/user_guide/hopsworks/jupyter.rst [hunk 2]
```diff
@@ -216,6 +216,92 @@ When you download a dataframe from spark to pandas with sparkmagic, it gives you
...
+repositories in GitHub. To authenticate in a remote service we use API keys which are safely stored in Hopsworks.
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/jupyter.rst]: @kouzant `Github ones. Authenticating against a remote service is done using API keys which...`

File: docs/user_guide/hopsworks/jupyter.rst [hunk 3]
```diff
@@ -216,6 +216,92 @@ When you download a dataframe from spark to pandas with sparkmagic, it gives you
...
+
+Getting an API key
+~~~~~~~~~~~~~~~~~~
+
+The first thing we need to do is to issue an API key from a remote hosting service. For the purpose of this guide it will be GitHub.
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/jupyter.rst]: @kouzant `to do is issue`

File: docs/user_guide/hopsworks/jupyter.rst [hunk 4]
```diff
@@ -216,6 +216,92 @@ When you download a dataframe from spark to pandas with sparkmagic, it gives you
...
+To do so, go to your **Settings** > **Developer Settings** > **Personal access tokens**
+
+Then click on **Generate new token**. Give a distinctive name to the token and select all repo scopes. Finally hit the **Generate token button**.
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/jupyter.rst]: @kouzant we should also include a link to Github's documentation how to do things. Chances are if they change their UI, we will forget to update the guide.

File: docs/user_guide/hopsworks/jupyter.rst [hunk 5]
```diff
@@ -216,6 +216,92 @@ When you download a dataframe from spark to pandas with sparkmagic, it gives you
...
+    :figclass: align-center
+
+    Starting JupyterLab with Git integration
+    
+Keep in mind that once you've enabled Git, you will **no longer be able** to see the notebooks stored in HDFS and vice versa. Notebooks
+versioned with Git will **not** be visible in Datasets browser. Another important note is that if you are running Jupyter Servers on Kubernetes
+and Git is enabled, notebooks are stored in the pod's local filesystem. So, if you stop Jupyter or the pod gets killed and you haven't **pushed**,
+your modifications will be lost.
+
+That’s the minimum configuration you should do. By default it will automatically **pull** from the default branch on Jupyter startup and **push**
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/jupyter.rst]: @kouzant `you should have.`

**Comment by kouzant** [comment]
Apart from the comments you made I should also change the instructions to reflect the new UI

**Pull request merged**